src/engine/sys/sdl_glimp.cpp

```diff
@@ -2154,6 +2156,217 @@ static void reportHardwareType( bool force )
+		// Mesa Nvidia for supported OpenGL 2+ hardware.
+		{ "nouveau", { glDriverVendor_t::MESA, glHardwareVendor_t::NVIDIA } },
+		// Proprietary Nvidia drivers on all systems like Linux, Windows, and macOS.
+		{ "NVIDIA Corporation", { glDriverVendor_t::NVIDIA, glHardwareVendor_t::NVIDIA } },
+		// Mesa Amber also provides "Nouveau", but this is for unsupported pre-OpenGL 2 Nvidia.
+	};
+
+	if ( vendorDriverHardware.count( glConfig.vendor_string ) )
+	{
+		auto& p = vendorDriverHardware.at( glConfig.vendor_string );
+		glConfig2.driverVendor = p.first;	
+		glConfig2.hardwareVendor = p.second;	
+		return;
+	}
+
+	// This vendor string is used by at least two different Intel drivers.
+	if ( !strcmp( glConfig.vendor_string, "Intel Corporation" ) )
+	{
+		if ( !strncmp( glConfig.renderer_string, "SWR ", 4 ) )
+		{
+			/* It is part of Mesa Amber, but was merged in Mesa lately after being an internal
+			Intel product for years. It doesn't share much things with Mesa, and we better
+			not assume it behaves the same as Mesa, it may even behave like Intel.
+			We keep glConfig2.driverVendor as glDriverVendor_t::UNKNOWN on purpose. */
+			glConfig2.hardwareVendor = glHardwareVendor_t::SOFTWARE;
+			return;
+		}
+		else
+		{
+			glConfig2.driverVendor = glDriverVendor_t::MESA;
+			glConfig2.hardwareVendor = glHardwareVendor_t::INTEL;
+			return;
+		}
+	}
+
+	// This vendor string is used by at least two different Microsoft drivers.
+	if ( !strcmp( glConfig.vendor_string, "Microsoft Corporation" ) )
+	{
+		/* The GL_VENDOR string can also be "Microsoft Corporation" with GL_RENDERER
+		set to "GDI Generic" which is not Mesa and isn't supported (OpenGL 1.1). */
+		if ( !strncmp( glConfig.renderer_string, "D3D12 ", 6 ) )
+		{
+			glConfig2.driverVendor = glDriverVendor_t::MESA;
+			// OpenGL over Direct3D12 translation.
+			glConfig2.hardwareVendor = glHardwareVendor_t::TRANSLATION;
+			return;
+		}
+	}
+
+	/* Those substrings at the beginnig of a renderer string are assumed to be unambiguous about
+	both driver and hardware. */
+	static const std::pair<std::string, glHardwareVendor_t> rendererMesaStartStringHardware[] = {
+		// "Mesa DRI R200" exists for ATI but isn't supported.
+		{ "Mesa DRI nv", glHardwareVendor_t::NVIDIA },
+		{ "Mesa DRI Intel(R)", glHardwareVendor_t::INTEL },
+	};
+
+	for ( auto& p : rendererMesaStartStringHardware )
+	{
+		if ( !strncmp( glConfig.renderer_string, p.first.c_str(), p.first.size() ) )
+		{
+			glConfig2.driverVendor = glDriverVendor_t::MESA;
+			glConfig2.hardwareVendor = p.second;
+			return;
+		}
+	}
+	
+	// Those vendor strings are assumed to be unambiguous about being from Mesa drivers.
+	static const std::string mesaVendors[] = {
+		"Mesa",
+		"Mesa Project",
+		"Mesa/X.org",
+		"X.Org",
+		// Zink.
+		"Collabora Ltd",
+		// virgl.
+		"Red Hat",
+		// llvmpipe, softpipe, SVGA3D.
+		"VMware, Inc.",
+	};
+
+	for ( auto& s : mesaVendors )
+	{
+		if ( !strcmp( glConfig.vendor_string, s.c_str() ) )
```

Comment:
When either side of the comparison is a std::string you can use `==`

Reply:
Done.